packages/datetime/src/dateInput.tsx

```diff
@@ -212,7 +212,14 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
         const dateValue = this.isMomentValidAndInRange(value) ? fromMomentToDate(value) : null;
         const dayPickerProps = {
             ...this.props.dayPickerProps,
-            onMonthChange: () => setTimeout(this.registerPopoverBlurHandler, 0),
+            // dom elements for the updated month is not available when
+            // onMonthChange is called. setTimeout is necessary to wait
+            // for the updated month to be rendered
+            onMonthChange: (month: Date) =>
+                setTimeout(() => {
+                    Utils.safeInvoke(this.props.dayPickerProps.onMonthChange, month);
```

Comment:
but this shouldn't be in the timeout, it should be called synchronously.

Reply:
- [ ] move `safeInvoke` out of `setTimeout`